src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy

```diff
@@ -151,6 +152,80 @@ sourceSets {
         buildResult.output.contains(':checkScalafixIntegTest SKIPPED')
     }
 
+    def 'scalafix semantic rule RemoveUnused fixes scala code'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FIXED_SCALA_CODE
+    }
+
+    def 'checkScalafix semantic rule RemoveUnused fails a build'() {
```

Comment:
sug.: 'checkScalafix should run semantic rewrite rule and fails the build without fixing input source files'